internal/saucecloud/xcuitest.go

```diff
@@ -176,37 +177,45 @@ func (r *XcuitestRunner) calculateJobsCount(suites []xcuitest.Suite) int {
 }
 
 // archiveAppsToIpaIfRequired checks if apps are a .ipa package. Otherwise, it generates one.
-func archiveAppsToIpaIfRequired(project *xcuitest.Project) (err error) {
-	appPath := project.Xcuitest.App
-	if !strings.HasSuffix(appPath, ".ipa") {
-		project.Xcuitest.App, err = archiveAppToIpa(appPath)
-		if err != nil {
-			log.Error().Msgf("Unable to archive %s to ipa: %v", appPath, err)
-			err = fmt.Errorf("unable to archive %s", appPath)
-			return
-		}
-	}
+func archiveAppsToIpaIfRequired(project *xcuitest.Project) error {
+	var err error
 	cache := map[string]string{}
+	project.Xcuitest.App, err = archiveAppToIpaIfRequired(project.Xcuitest.App, &cache)
+	if err != nil {
+		return err
+	}
+
 	for i, s := range project.Suites {
-		if strings.HasSuffix(s.TestApp, ".ipa") {
-			continue
+		project.Suites[i].TestApp, err = archiveAppToIpaIfRequired(s.TestApp, &cache)
+		if err != nil {
+			return err
 		}
+	}
+	return nil
+}
 
-		if val, ok := cache[s.TestApp]; ok {
-			project.Suites[i].TestApp = val
-			continue
-		}
+func archiveAppToIpaIfRequired(appPath string, cache *map[string]string) (string, error) {
```

Comment:
Slices and maps don't need to be passed as pointers. Both structures **use pointers** for the values they carry.
This simplifies the handling.

```suggestion
func archiveAppToIpaIfRequired(appPath string, cache map[string]string) (string, error) {
```